src/gui/mzroll/spectrawidget.cpp

```diff
@@ -394,33 +396,79 @@ void SpectraWidget::setTitle()
         return;
     }
 
-    QString sampleName;
+    QString sampleName("");
     if (_currentScan->sample)  sampleName = QString(_currentScan->sample->sampleName.c_str());
-
-    _titleText += tr("<b>%1</b>").arg(sampleName);
     
     if (_currentScan->scannum)
-        _titleText += tr(" Scan#<b>%1</b>  ").arg(QString::number(_currentScan->scannum));
+        _titleText += tr("<b>Scan#</b> %1  ").arg(QString::number(_currentScan->scannum));
 
     if (_currentScan->rt)
-        _titleText += tr("Rt:<b>%1</b>  ").arg(QString::number(_currentScan->rt, 'f', 2));
+        _titleText += tr("<b>Rt:</b> %1  ").arg(QString::number(_currentScan->rt, 'f', 2));
 
     if (_currentScan->mslevel)
-        _titleText += tr("MS Level:<b>%1</b>  ").arg(QString::number(_currentScan->mslevel));
+        _titleText += tr("<b>MS Level:</b> %1  ").arg(QString::number(_currentScan->mslevel));
 
     if (_currentScan->precursorMz) {
-        _titleText += tr("Pre m/z:<b>%1</b>  ").arg(QString::number(_currentScan->precursorMz,'f',4));
+        _titleText += tr("<b>Pre m/z:</b> %1  ").arg(QString::number(_currentScan->precursorMz,'f',4));
     }
 
     if (_currentScan->collisionEnergy) {
-        _titleText += tr("CE:<b>%1</b>  ").arg(QString::number(_currentScan->collisionEnergy,'f',0));
+        _titleText += tr("<b>CE:</b> %1  ").arg(QString::number(_currentScan->collisionEnergy,'f',0));
     }
 
     if (_currentScan->productMz) {    
-       _titleText += tr("Prod m/z:<b>%1</b>  ").arg(QString::number(_currentScan->productMz,'f',3));
+       _titleText += tr("<b>Prod m/z:</b> %1  ").arg(QString::number(_currentScan->productMz,'f',3));
+    }
+
+    if (_currentScan->precursorMz > 0 && _currentScan->sample) {
```

Comment:
Why is the second check required ?

Reply:
purity is not a pre-calculated value. It's calculated from the sample. If `_currentScan` does not have an associated sample (this can happen in case of fragmentation widget), purity cannot be calculated so it's checking for that possibility.

Reply:
Can remove the sample check. PrecursorMz check is there to determine if it's MS2 or not but the sample one isn't really necessary.